src/main/java/gregtech/api/multipart/MultipartPipeLike.java

```diff
@@ -0,0 +1,361 @@
+    private Q baseProperty;
+    private P tileProperty;
+    private int color;
+    private int internalConnection = 0;
+    private int renderMask = 0;
+
+    private Cuboid6 centerBox;
+    private List<Cuboid6> sideBoxes = new ArrayList<>();
+
+    public MultipartPipeLike(PipeLikeObjectFactory<Q, P, T, C> factory) {
+        this.factory = factory;
+    }
+
+    @SuppressWarnings("unchecked")
+    public MultipartPipeLike(PipeLikeObjectFactory<Q, P, T, C> factory, TileEntityPipeLike<Q, P, T, C> tile, IBlockState state) {
+        this.factory = factory;
+        block = (BlockPipeLike<Q, P, T, C>) state.getBlock();
+        baseProperty = state.getValue(block.getBaseProperty());
+        tileProperty = block.getActualProperty(baseProperty);
+        color = tile.getColor();
+        reinitShape();
+    }
+
+    /////////////////////////////// BASE PROPERTIES ////////////////////////////////////////
+
+    @Override
+    public void save(NBTTagCompound tag) {
+        tag.setString("material", block.material.toString());
+        tag.setInteger("baseProperty", baseProperty.ordinal());
+        tag.setInteger("color", color);
+        tag.setInteger("internalConnection", internalConnection);
+        tag.setInteger("renderMask", renderMask);
+    }
+
+    @Override
+    public void load(NBTTagCompound tag) {
+        block = factory.getBlock(Material.MATERIAL_REGISTRY.getObject(tag.getString("material")));
+        baseProperty = factory.getBaseProperty(tag.getInteger("baseProperty"));
+        tileProperty = block.getActualProperty(baseProperty);
+        color = tag.getInteger("color");
+        internalConnection = tag.getInteger("internalConnection");
+        renderMask = tag.getInteger("renderMask");
+        reinitShape();
+    }
+
+    @Override
+    public ResourceLocation getType() {
+        return factory.multipartType;
+    }
+
+    @Override
+    public World getWorld() {
+        return world();
+    }
+
+    @Override
+    public BlockPos getPos() {
+        return pos();
+    }
+
+    @Override
+    public Material getMaterial() {
+        return block.material;
+    }
+
+    @Override
+    public Q getBaseProperty() {
+        return baseProperty;
+    }
+
+    @Override
+    public P getTileProperty() {
+        return tileProperty;
+    }
+
+    @Override
+    public int getColor() {
+        return color;
+    }
+
+    @Override
+    public void setColor(int color) {
+        this.color = color;
+        updateRenderMask(true);
+        onConnectionUpdate();
+    }
+
+    @Override
+    public int getInternalConnections() {
+        return internalConnection;
+    }
+
+    ///////////////////////////////// ITEM DROPPED /////////////////////////////////////////
+
+    protected ItemStack getItem() {
+        return block.getItem(baseProperty);
+    }
+
+    @Override
+    public Iterable<ItemStack> getDrops() {//TODO Tools
+        return Collections.singleton(getItem());//TODO Covers?
```

Comment:
covers dropping is handled entirely by multipart, only thing you need to override for tools is getStrength(EntityPlayer, CuboidRayTraceResult).

Reply:
what do you mean by covers? I thought you meant FMP covers. IF you're talking about some GT-ish covers, i don't really seem them needed anyhow.

Reply:
I mean GT-ish covers. Well, just ignore it.